File: app/src/main/java/br/com/amilton/adapter/StatementAdapter.java
package br.com.amilton.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;
import java.util.List;

import br.com.amilton.R;
import br.com.amilton.databinding.StatementRowBinding;
import br.com.amilton.model.Statement;

public class StatementAdapter extends RecyclerView.Adapter<StatementAdapter.StatementViewHolder> {

    private List<Statement> list = new ArrayList<>();

    @NonNull
    @Override
    public StatementViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new StatementViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.statement_row, parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull StatementViewHolder holder, int position) {
        Statement statements = list.get(position);
        holder.statementRowBinding.setStatement(statements);
        holder.statementRowBinding.executePendingBindings();
    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    public void setList(List<Statement> list) {
        this.list = list;
        notifyDataSetChanged();
    }

    static class StatementViewHolder extends RecyclerView.ViewHolder {
        private final StatementRowBinding statementRowBinding;
        StatementViewHolder(View v) {
            super(v);
            statementRowBinding = StatementRowBinding.bind(v);
        }
    }
}

File: app/src/main/java/com/gustavo/bankandroid/statements/ui/adapter/StatementAdapter.kt
package com.gustavo.bankandroid.statements.ui.adapter

import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.gustavo.bankandroid.entity.UserStatementItem

class StatementAdapter() : RecyclerView.Adapter<StatementViewHolder>() {

    var statementList: List<UserStatementItem> = emptyList()
        set(value) {
            field = value
            notifyDataSetChanged()
        }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): StatementViewHolder =
        StatementViewHolder(parent)

    override fun getItemCount(): Int = statementList.size

    override fun onBindViewHolder(holder: StatementViewHolder, position: Int) {
        holder.bind(statementList[position])
    }
}